src/buildout/wheel/__init__.py

```diff
@@ -1,56 +1,119 @@
-import distlib.wheel
-import humpty
 import os
+import os.path
+import shutil
+import sys
 import pkg_resources
 import setuptools.package_index
 from six.moves.urllib import parse
 import wheel.install
+import wheel.decorator
 import pip.pep425tags
 import zc.buildout.easy_install
+import distutils.command.install
 
-orig_interpret_distro_name = setuptools.package_index.interpret_distro_name
+NAMESPACE_STUB_PATH = os.path.join(os.path.dirname(__file__), 'namespace_stub.py')
+assert os.path.isfile(NAMESPACE_STUB_PATH)
+
+orig_distros_for_location = setuptools.package_index.distros_for_location
 original_wheel_to_egg = zc.buildout.easy_install.wheel_to_egg
 
-wheel_supported = pip.pep425tags.get_supported
+class SelfDestructingEgg(pkg_resources.Distribution):
+    """Since the location for creating the egg is inside the download directory
+    We must self destruct to avoid having both the wheel and the generated egg
+    there at the same time."""
+    def __del__(self):
+        shutil.rmtree(self.location)
```

Comment:
This delays the cleanup indefinitely if you use PyPy, and possibly never does it at all.

Is there no better place to do the cleanup deterministically?